### PR TITLE
Pie Chart Panel

### PR DESCRIPTION
**Pull request by Rajat-Dabade**
Loom: https://www.loom.com/share/a854d088ef044dd9a11c1e673be7dd81?sid=968a81ec-df94-4a1f-bd7e-92c85008bb8f

### CODE REVIEW

**Review by YounixM — DISMISSED** [review]

**Comment by pranay01** [comment]
@srikanthccv Does having support for multiple queries in case of Pie chart make sense? Is there a use case for that?

Also, not clear on how should legends work here. e.g If we use a single fixed word for legend

File: frontend/src/container/PanelWrapper/PiePanelWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+								style={{
+									...tooltipStyles,
+									background: isDarkMode ? Color.BG_INK_400 : Color.BG_VANILLA_100,
+									color: isDarkMode ? Color.BG_VANILLA_100 : Color.BG_INK_400,
+								}}
+							>
+								<div
+									style={{
+										background: tooltipData.color,
+										width: '15px',
```
Comment (YounixM) [inline frontend/src/container/PanelWrapper/PiePanelWrapper.tsx:179]: Can move these styles to css file.

File: frontend/src/container/PanelWrapper/PiePanelWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+										height: '3px',
+										borderRadius: '2px',
+									}}
+								/>
+								{tooltipData.key}
+								<div>
```
Comment (YounixM) [inline frontend/src/container/PanelWrapper/PiePanelWrapper.tsx:185]: We can use div with a className and can get rid of strong.

File: frontend/src/container/PanelWrapper/PiePanelWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+									}}
+									padAngle={0.02}
+									cornerRadius={3}
+									width={size}
+									height={size}
+								>
+									{
+										// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+										(pie) =>
+											pie.arcs.map((arc, index) => {
```
Comment (YounixM) [inline frontend/src/container/PanelWrapper/PiePanelWrapper.tsx:117]: If we have used this code from another source, can we link it here for future reference?

File: frontend/src/container/PanelWrapper/PiePanelWrapper.tsx (hunk 1 above)
Comment (Rajat-Dabade) [reply frontend/src/container/PanelWrapper/PiePanelWrapper.tsx:179]: Sure.

File: frontend/src/container/PanelWrapper/PiePanelWrapper.tsx (hunk 2 above)
Comment (Rajat-Dabade) [reply frontend/src/container/PanelWrapper/PiePanelWrapper.tsx:185]: Done,

**Comment by srikanthccv** [comment]
> Does having support for multiple queries in case of Pie chart make sense? Is there a use case for that?

Yes, error rate by service.

> not clear on how should legends work here

Each pie of the chart if formatted according to the legend setting.

File: frontend/src/container/PanelWrapper/PiePanelWrapper.tsx (hunk 3 above)
Comment (Rajat-Dabade) [reply frontend/src/container/PanelWrapper/PiePanelWrapper.tsx:118]: Added

**Review by YounixM — DISMISSED** [review]

**Review by YounixM — APPROVED** [review]

**Pull request merged**